agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -653,12 +656,62 @@ bool monitor_thread::update_ap_stats()
     return true;
 }
 
+bool monitor_thread::handle_ap_metrics_query(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    const auto mid            = cmdu_rx.getMessageId();
+    auto ap_metrics_query_tlv = cmdu_rx.getClass<wfa_map::tlvApMetricQuery>();
+
+    if (!ap_metrics_query_tlv) {
+        LOG(ERROR) << "AP Metrics Query CMDU mid=" << mid << " does not have AP Metric Query TLV";
+        return false;
+    }
+    for (int bssid_idx = 0; bssid_idx < ap_metrics_query_tlv->bssid_list_length(); bssid_idx++) {
+        auto bssid = ap_metrics_query_tlv->bssid_list(bssid_idx);
+        if (!std::get<0>(bssid)) {
+            LOG(ERROR) << "Failed to get bssid " << bssid_idx << " from AP_METRICS_QUERY";
+            //return false;
+        }
+        LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE, mid=" << std::hex << int(mid)
+                   << "  bssid " << std::get<1>(bssid);
+    }
+
+    auto cmdu_header = cmdu_tx.create(mid, ieee1905_1::eMessageType::AP_METRICS_RESPONSE_MESSAGE);
+
+    if (!cmdu_header) {
+        LOG(ERROR) << "Failed building IEEE1905 AP_METRICS_RESPONSE_MESSAGE";
+        return false;
+    }
+
+    auto ap_metrics_response_tlv = cmdu_tx.addClass<wfa_map::tlvApMetric>();
+    LOG(DEBUG) << "Sending AP_METRICS_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+    return message_com::send_cmdu(slave_socket, cmdu_tx);
+}
+
+bool monitor_thread::handle_cmdu_control_ieee1905_1_message(Socket *sd,
+                                                            ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto cmdu_message_type = cmdu_rx.getMessageType();
+
+    if (!sd) {
+        LOG(ERROR) << "Socket error";
+        return false;
+    }
+
+    switch (cmdu_message_type) {
+    case ieee1905_1::eMessageType::AP_METRICS_QUERY_MESSAGE:
+        return handle_ap_metrics_query(sd, cmdu_rx);
+    default:
+        LOG(ERROR) << "Unknown CMDU message type: " << std::hex << int(cmdu_message_type);
+        return false;
+    }
+}
+
 bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
     auto beerocks_header = message_com::parse_intel_vs_message(cmdu_rx);
     if (beerocks_header == nullptr) {
-        LOG(ERROR) << "Not a vendor specific message";
-        return false;
+        LOG(DEBUG) << "Received IEEE 1905.1 message";
+        handle_cmdu_control_ieee1905_1_message(sd, cmdu_rx);
```

Comment:
I've also had to implement this sort of differentiating and dispatching vendor-specific messages and IEEE 1905.1 messages. However, I've done it in a slightly different way, by checking the type of message. At this point we'll have a merge conflict and we should agree on which mechanism should be used.

Reply:
@mariomaz If you think your mechanism is better, please make a quick PR with only that commit, and Vlad can rebase on it.

Reply:
Done, see #1236.
New mechanism is also implemented in #1233

Reply:
So we do not have conflicts, @mariomaz please check updated changes